src/tribol/interface/tribol.cpp

```diff
@@ -391,6 +408,107 @@ void registerMesh( integer meshId,
 
 } // end of registerMesh()
 
+//------------------------------------------------------------------------------
+void registerMfemMesh( integer cs_id,
+                      integer mesh_id_1,
+                      integer mesh_id_2,
+                      mfem::ParMesh& mesh,
+                      const mfem::ParGridFunction& current_coords,
+                      const std::set<integer>& attributes_1,
+                      const std::set<integer>& attributes_2,
+                      integer contact_mode,
+                      integer contact_case,
+                      integer contact_method,
+                      integer contact_model,
+                      integer enforcement_method,
+                      integer binning_method)
+{
+   std::unique_ptr<mfem::FiniteElementCollection> dual_fec = nullptr;
+   integer dual_vdim = 0;
+   auto mfem_data = std::make_unique<MfemMeshData>(
+      mesh_id_1,
+      mesh_id_2,
+      mesh,
+      current_coords,
+      attributes_1,
+      attributes_2
+   );
+   // register empty meshes so the coupling scheme is valid
+   registerMesh(
+      mesh_id_1, 0, 0, nullptr, mfem_data->GetElemType(), nullptr, nullptr);
+   registerMesh(
+      mesh_id_2, 0, 0, nullptr, mfem_data->GetElemType(), nullptr, nullptr);
+   registerCouplingScheme(
+      cs_id,
+      mesh_id_1,
+      mesh_id_2,
+      contact_mode,
+      contact_case,
+      contact_method,
+      contact_model,
+      enforcement_method,
+      binning_method
+   );
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   if (enforcement_method == LAGRANGE_MULTIPLIER)
+   {
+      dual_fec = std::make_unique<mfem::H1_FECollection>(
+         current_coords.FESpace()->FEColl()->GetOrder(),
+         mesh.SpaceDimension()
+      );
+      if (contact_model == FRICTIONLESS)
+      {
+         dual_vdim = 1;
+      }
+      else if (contact_model == TIED || contact_model == COULOMB)
+      {
+         dual_vdim = mesh.SpaceDimension();
+      }
+      else
+      {
+         SLIC_ERROR_ROOT("Unsupported contact model. "
+           "Only FRICTIONLESS, TIED, and COULOMB supported.");
+      }
+      coupling_scheme->setMfemSubmeshData(
```

Comment:
Add comment as to what this is doing. This is just complicated enough some clarity would be helpful. Looks like we get the submesh, get the LOR submesh, etc. to construct a pointer to submesh data?

Reply:
Added.  This basically creates the submesh to redecomp transfer operators and creates a pressure field on the submesh.